### PR TITLE
Consolidate public room search experience

### PR DESCRIPTION
**Pull request by gsouquet**
Fixes https://github.com/vector-im/element-web/issues/22846

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Consolidate public room search experience ([\#9605](https://github.com/matrix-org/matrix-react-sdk/pull/9605)). Fixes vector-im/element-web#22846.

### CODE REVIEW

File: src/Rooms.ts
Comment (turt2live) [inline src/Rooms.ts:42]: this appears to be a new tsc --strict failure, fwiw

File: cypress/e2e/room-directory/room-directory.spec.ts
Comment (turt2live) [inline cypress/e2e/room-directory/room-directory.spec.ts:98]: we can drop the `as` if we're not using it?

Comment (turt2live) [reply cypress/e2e/room-directory/room-directory.spec.ts:98]: (we should probably be using it though?)

**Review by kittykat — APPROVED** [review]
Makes sense from a product POV

**Comment by gsouquet** [comment]
@turt2live Looks like percy's test need review, and approval too. I do not have an account and can't change those.

**Comment by t3chguy** [comment]
I've approved the diffs which look sane, though a bunch of auth modals seem to have changed size which is worrying, and the changed snapshots for the room directory do not include the modal wrapper which makes the diff less visually useful - once those are addressed I will approve the Percy diff

File: src/utils/rooms.ts
Comment (MadLittleMods) [inline src/utils/rooms.ts:135]: This PR removed all instances of `GenericError`, we could also remove the class in general.

Removed in https://github.com/matrix-org/matrix-react-sdk/pull/10500

**Pull request merged**
